clcache.py

```diff
@@ -747,14 +747,32 @@ def _preprocessToStdout(options):
         return ('E' in options or 'EP' in options) and 'P' not in options
 
     @staticmethod
+    def _parsePathArgument(options, argumentName, sourceFile, extension):
+        if argumentName in options:
+            # Handle user input
+            outputFile = options[argumentName][0]
+            outputFile = os.path.normpath(outputFile)
+
+            if os.path.isdir(outputFile):
+                outputFile = os.path.join(outputFile, basenameWithoutExtension(sourceFile) + extension)
+        else:
+            # Generate from .c/.cpp filename
+            outputFile = basenameWithoutExtension(sourceFile) + extension
+        return outputFile
```

Comment:
Okay, I don't really feel strongly about this, so let's go with what you propose.